File: ts-plugins/src/docs/flow-documentation-plugin.ts
import type { FlowDefinition } from '@unlaxer/tramli';
import type { DocumentationPlugin as DocPluginSPI, PluginDescriptor } from '../api/types.js';
import { DocumentationPlugin } from './documentation-plugin.js';

export class FlowDocumentationPlugin<S extends string> implements DocPluginSPI<FlowDefinition<S>> {
  private readonly delegate = new DocumentationPlugin();

  descriptor(): PluginDescriptor {
    return {
      id: 'docs',
      displayName: 'Documentation Generator',
      description: 'Renders a markdown flow catalog from a flow definition.',
    };
  }
  kind() { return 'DOCUMENTATION' as const; }

  generate(input: FlowDefinition<S>): string {
    return this.delegate.toMarkdown(input);
  }
}

File: lang/java-plugins/src/main/java/org/unlaxer/tramli/plugins/docs/FlowDocumentationPlugin.java
package org.unlaxer.tramli.plugins.docs;

import org.unlaxer.tramli.FlowDefinition;
import org.unlaxer.tramli.FlowState;
import org.unlaxer.tramli.plugins.api.PluginDescriptor;

public final class FlowDocumentationPlugin<S extends Enum<S> & FlowState> implements org.unlaxer.tramli.plugins.api.DocumentationPlugin<FlowDefinition<S>> {
    private final DocumentationPlugin delegate = new DocumentationPlugin();

    @Override
    public PluginDescriptor descriptor() {
        return new PluginDescriptor("docs", "Documentation Generator", "Renders a markdown flow catalog from a flow definition.");
    }

    @Override
    public String generate(FlowDefinition<S> input) {
        return delegate.toMarkdown(input);
    }
}
